### DISCUSSION TITLE
how to train an NEL model with transformers?

### DISCUSSION THREAD

**Discussion by moyid** [explosion/spaCy#7315] [Help: Coding & Implementations] [answered]
Hi I trained an NER model with transformers, and now I want to add entity linking to it. The problem is that the NER model I trained doesn't return any vector representations of text, and for NEL I need to create a knowledge base, where you need to add the entity_vector. 

when I run this code:
```
vectors_model = "path/to/transformers/NER"
nlp = spacy.load(Path(vectors_model))
kb = KnowledgeBase(vocab=nlp.vocab, entity_vector_length=300)

for qid, names in name_dict.items():
    desc = ' '.join(names)
    desc_doc = nlp(desc)
    desc_enc = desc_doc.vector
    kb.add_entity(entity=qid, entity_vector=desc_enc, freq=342)
```

I get the error:
`ValueError: [E141] Entity vectors should be of length 300 instead of the provided 0.`

**Comment by moyid** [comment]
following up on this, any suggestions? Thank you!

**Comment by svlandeg** [comment, marked as answer]
Hi @moyid, a related discussion thread is here: https://github.com/explosion/spaCy/discussions/6511. That thread discusses how to set up user hooks to derive the `vector` property from the tensors that are being produced by the transformer in your pipeline. Hope that helps!

**Comment by nkaramooz** [comment]
@moyid were you able to get this working? I'm still struggling despite reviewing the related thread.
